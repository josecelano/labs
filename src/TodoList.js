/**
 * This component should manage the todo items, letting users check existing ones and add new ones
 */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import TodoItem from './TodoItem';
import TodoInput from './TodoInput';
import {
	checkItem,
	addItem,
	editInput,
} from './actions';

export class TodoList extends Component {
	// Check an item
	checkItem = (text) => {
		this.props.dispatch(checkItem(text));
	}

	// Add an item
	addItem = (evt) => {
		evt.preventDefault();
		this.props.dispatch(addItem());
	}

	// Edit the input
	editInput = (evt) => {
		this.props.dispatch(editInput(evt.target.value));
	}

	render() {
		return (
			<div>
				<ul>
					{this.props.items.map((item, index) => (
						<li key={index}>
							<TodoItem
								onClick={this.checkItem}
								text={item.text}
								checked={item.checked}
							/>
						</li>
					))}
				</ul>
				<form onSubmit={this.addItem}>
					<TodoInput
						onChange={this.editInput}
						value={this.props.value}
					/>
				</form>
			</div>
		)
	}
}

const mapStateToProps = (state) => {
	return {
		items: state.items,
		inputValue: state.inputValue,
	};
};

export default connect(mapStateToProps)(TodoList);
