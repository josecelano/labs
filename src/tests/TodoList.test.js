/**
 * Tests for the TodoList
 */
import React from 'react';
import renderer from 'react-test-renderer';
import { TodoList } from '../TodoList';
import { shallow } from 'enzyme'; 

describe('<TodoList />', () => {
    it('should render an todo list', () => {
        const tree = renderer
            .create(<TodoList items={ [] }/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})