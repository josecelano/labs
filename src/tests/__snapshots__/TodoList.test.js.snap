exports[`<TodoList /> should render an todo list 1`] = `
<div>
  <ul />
  <form
    onSubmit={[Function]}>
    <input
      onChange={[Function]}
      type="text"
      value={undefined} />
  </form>
</div>
`;
